.github/workflows/check-package-builds.yml

```diff
@@ -0,0 +1,166 @@
+            build-prep: |
+              license_file_name=$(grep -oP '^%license\s+\K\S+' "$REGULAR_PKG_SPEC_PATH")
+              if [[ -z "$license_file_name" ]]; then
+                echo "ERROR: no license file found in the spec $REGULAR_PKG_SPEC_PATH"
+                exit 1
+              fi
+              # Tagging a license file as a documentation file will not fail the license check on the 'warn' level.
+              sed -i "/^%license/a %doc $license_file_name" "$REGULAR_PKG_SPEC_PATH"
+
+          - check-name: "Fatal license check succeeds on duplicated license as documentation"
+            package-type: "REGULAR_PKG"
+            extra-args: "LICENSE_CHECK_MODE=fatal"
+            build-prep: |
+              license_file_name=$(grep -oP '^%license\s+\K\S+' "$REGULAR_PKG_SPEC_PATH")
+              if [[ -z "$license_file_name" ]]; then
+                echo "ERROR: no license file found in the spec $REGULAR_PKG_SPEC_PATH"
+                exit 1
+              fi
+              # Tagging a license file as a documentation file will not fail the license check on the 'fatal' level.
+              sed -i "/^%license/a %doc $license_file_name" "$REGULAR_PKG_SPEC_PATH"
+
+          - check-name: "Fatal license check fails"
+            package-type: "REGULAR_PKG"
+            error-pattern: "Number of SRPMs with license errors:\\s+1\\s*$"
+            extra-args: "LICENSE_CHECK_MODE=fatal"
+            build-prep: |
+              if ! grep -q '^%license' "$REGULAR_PKG_SPEC_PATH"; then
+                echo "ERROR: no '%license' macro found in the spec $REGULAR_PKG_SPEC_PATH"
+                exit 1
+              fi
+              # Tagging a license file as a documentation file will cause the license check to fail.
+              sed -i "s/^%license/%doc/" "$REGULAR_PKG_SPEC_PATH"
+
+          - check-name: "Pedantic license check fails"
+            package-type: "REGULAR_PKG"
+            error-pattern: "Number of SRPMs with license errors:\\s+1\\s*$"
+            extra-args: "LICENSE_CHECK_MODE=pedantic"
+            build-prep: |
+              license_file_name=$(grep -oP '^%license\s+\K\S+' "$REGULAR_PKG_SPEC_PATH")
+              if [[ -z "$license_file_name" ]]; then
+                echo "ERROR: no license file found in the spec $REGULAR_PKG_SPEC_PATH"
+                exit 1
+              fi
+              sed -i "/^%license/a %doc $license_file_name" "$REGULAR_PKG_SPEC_PATH"
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Checkout a stable version of the specs
+        uses: ./.github/actions/checkout-with-stable-pkgs
+
+      - name: Prepare the build environment
+        if: ${{ matrix.build-prep != '' }}
+        run: |
+          set -euo pipefail
+
+          ${{ matrix.build-prep }}
+
+      - name: Run the build
+        run: |
+          set -euo pipefail
+
+          if sudo make -C toolkit -j$(nproc) build-packages \
+            PACKAGE_REBUILD_LIST="${{ env[matrix.package-type] }}" \
```

Comment:
Do we have documentation somewhere about what package(s) are actually build during these tests? in the wiki somewhere maybe?